x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/integration/DataLifecycleServiceRuntimeSecurityIT.java

```diff
@@ -0,0 +1,274 @@
+            assertThat(
+                indicesAndErrors.values(),
+                hasItem(allOf(containsString("security_exception"), containsString("unauthorized for user [_dlm]")))
+            );
+        });
+    }
+
+    public void testRolloverAndRetentionWithSystemDataStreamAuthorized() throws Exception {
+        String dataStreamName = SystemDataStreamTestPlugin.SYSTEM_DATA_STREAM_NAME;
+        indexDoc(dataStreamName);
+
+        assertBusy(() -> {
+            assertNoAuthzErrors();
+            List<Index> backingIndices = getDataStreamBackingIndices(dataStreamName);
+            assertThat(backingIndices.size(), equalTo(1));
+            // we expect the data stream to have only one backing index, the write one, with generation 2
+            // as generation 1 would've been deleted by DLM given the lifecycle configuration
+            String writeIndex = backingIndices.get(0).getName();
+            assertThat(writeIndex, backingIndexEqualTo(dataStreamName, 2));
+        });
+    }
+
+    private Map<String, String> collectErrorsFromStoreAsMap() {
+        Iterable<DataLifecycleService> lifecycleServices = internalCluster().getInstances(DataLifecycleService.class);
+        Map<String, String> indicesAndErrors = new HashMap<>();
+        for (DataLifecycleService lifecycleService : lifecycleServices) {
+            DataLifecycleErrorStore errorStore = lifecycleService.getErrorStore();
+            List<String> allIndices = errorStore.getAllIndices();
+            for (var index : allIndices) {
+                indicesAndErrors.put(index, errorStore.getError(index));
+            }
+        }
+        return indicesAndErrors;
+    }
+
+    private void prepareDataStreamAndIndex(String dataStreamName, DataLifecycle lifecycle) throws IOException, InterruptedException,
+        ExecutionException {
+        putComposableIndexTemplate("id1", null, List.of(dataStreamName + "*"), null, null, lifecycle);
+        CreateDataStreamAction.Request createDataStreamRequest = new CreateDataStreamAction.Request(dataStreamName);
+        client().execute(CreateDataStreamAction.INSTANCE, createDataStreamRequest).get();
+        indexDoc(dataStreamName);
+    }
+
+    private List<Index> getDataStreamBackingIndices(String dataStreamName) {
+        GetDataStreamAction.Request getDataStreamRequest = new GetDataStreamAction.Request(new String[] { dataStreamName });
+        GetDataStreamAction.Response getDataStreamResponse = client().execute(GetDataStreamAction.INSTANCE, getDataStreamRequest)
+            .actionGet();
+        assertThat(getDataStreamResponse.getDataStreams().size(), equalTo(1));
+        assertThat(getDataStreamResponse.getDataStreams().get(0).getDataStream().getName(), equalTo(dataStreamName));
+        return getDataStreamResponse.getDataStreams().get(0).getDataStream().getIndices();
+    }
+
+    private void assertNoAuthzErrors() {
+        var indicesAndErrors = collectErrorsFromStoreAsMap();
```

Comment:
Started with an assertion that there are no errors at all, however, force merge seems flakey and sometimes fails with:

"Force merge request only had 1 successful shards out of a total of 2"

Example [failure](https://gradle-enterprise.elastic.co/s/k6qjhpgry5vou/tests/:x-pack:plugin:security:internalClusterTest/org.elasticsearch.integration.DataLifecycleServiceRuntimeSecurityIT/testRolloverLifecycleAndForceMergeAuthorized?top-execution=1).

This looks like a separate issue, not related to privileges, since it occurs even when I assign `all` on everything to the internal user. As such, just asserting that there are no security related errors here instead, to keep things moving.